test/integration/buffer_filter_integration_test.cc

```diff
@@ -0,0 +1,41 @@
+#include "test/integration/protocol_integration.h"
+
+namespace Envoy {
+
+typedef ProtocolIntegrationTest BufferIntegrationTest;
+
+typedef ProtocolIntegrationTest OtherIntegrationTest;
+
+INSTANTIATE_TEST_CASE_P(Protocols, BufferIntegrationTest,
+                        testing::ValuesIn(ProtocolIntegrationTest::GetProtocolTestParams()),
+                        ProtocolIntegrationTest::protocolTestParamsToString);
+
+TEST_P(BufferIntegrationTest, RouterNotFoundBodyBuffer) {
+  config_helper_.addFilter(ConfigHelper::DEFAULT_BUFFER_FILTER);
+  testRouterNotFoundWithBody();
+}
+
+TEST_P(BufferIntegrationTest, RouterRequestAndResponseWithGiantBodyBuffer) {
+  config_helper_.addFilter(ConfigHelper::DEFAULT_BUFFER_FILTER);
+  testRouterRequestAndResponseWithBody(4 * 1024 * 1024, 4 * 1024 * 1024, false);
+}
+
+TEST_P(BufferIntegrationTest, RouterHeaderOnlyRequestAndResponseBuffer) {
+  config_helper_.addFilter(ConfigHelper::DEFAULT_BUFFER_FILTER);
+  testRouterHeaderOnlyRequestAndResponse(true);
+}
+
+TEST_P(BufferIntegrationTest, RouterRequestAndResponseWithBodyBuffer) {
+  config_helper_.addFilter(ConfigHelper::DEFAULT_BUFFER_FILTER);
+  testRouterRequestAndResponseWithBody(1024, 512, false);
+}
+
+TEST_P(BufferIntegrationTest, RouterRequestAndResponseWithZeroByteBodyBuffer) {
+  config_helper_.addFilter(ConfigHelper::DEFAULT_BUFFER_FILTER);
+  testRouterRequestAndResponseWithBody(0, 0, false);
+}
+
+// Verifies the typedef saves us from all ProtocolIntergrationTests from being rerun.
+TEST_P(OtherIntegrationTest, Foo) { ASSERT(0); }
```

Comment:
nit: `ASSERT(false);`

I'm also a little confused about what this is doing. Can you expand the comment? And if this should be done in every instantiation of this test should we put in the fixture usage docs?

Reply:
just use FAIL() from gtest :) ASSERT will be compiled out in release builds.

Reply:
Oops - that was supposed to have a TODO to remove.
tl;dr: will remove 

That was me making sure my typedef workaround worked.  I'm used to having one big test framework and each file has TEST_F(Foo, BarTest).  With Envoy we could do TEST_P(ProtocolngegrationTest, BarTest) and INSTANTIATE (ProtocolIntegrationTest) but I'm fairly well convinced that would play horribly badly with the giant ubertest for coverage.  We'd have 15 identical instances of INSTANTIATE-all-the-protocol-tests and everything would run 15 times. Yuck.

I wanted to make sure if we typedef both BufferIntegrationTest and OtherIntegrationTest to ProtocolTest and then INSTANTIATED the BufferIntegrationTest it would avoid the double instantiation problem.

Some day we won't have the ubertest and everything will be easier :-)